### PR TITLE
optionally generate location theme distribution maps

### PR DESCRIPTION
**Pull request by IsaackMwenda**
This adds an optional boolean to generate location theme distribution maps on demand.

### CODE REVIEW

File: automated_analysis.py [hunk 1]
```diff
@@ -382,50 +382,54 @@ def set_survey_percentages(survey_counts, total_survey_counts):
...
-            # Plot maps of each of the normal themes for this coding configuration.
-            map_index = 2  # (index 1 was used in the total relevant map's filename).
-            for code in cc.code_scheme.codes:
-                if code.code_type != CodeTypes.NORMAL:
-                    continue
-
-                theme = f"{cc.analysis_file_key}_{code.string_value}"
-                log.info(f"Generating a map of per-region participation for {theme}...")
-                demographic_counts = episode[theme]
+    generate_region_theme_distribution_maps = False #TODO: move this to config file?
```
Comment (as2388) [inline automated_analysis.py]: Yes please, or at least to the top of this script for now until we figure out the best way to make automated analysis easily configurable.

**Review by as2388 — COMMENTED** [review]
What's the motivation for doing this?

File: automated_analysis.py (hunk 1 above)
Comment (as2388) [inline automated_analysis.py]: If this is optional, you'll want to make the upload optional too, or clear the automated analysis folder between runs

**Comment by IsaackMwenda** [comment]
> What's the motivation for doing this?

The files are not being used at the moment but might be requested. Instead of deprecating the code, it's better to keep it as an optional argument in configs so that we can easily run on demand.

File: automated_analysis.py (hunk 1 above)
Comment (IsaackMwenda) [reply automated_analysis.py]: will update this in this pr https://github.com/AfricasVoices/Project-IMAQAL-COVID19/pull/50 now that we have automated analysis configs in configuration files. Thanks

Comment (IsaackMwenda) [reply automated_analysis.py]: moved automated analysis configs to configuration files. thanks

File: configuration/pipeline_config.json [hunk 2]
```diff
@@ -122,6 +122,11 @@
   "ProjectEndDate": "2100-01-01T00:00:00+03:00",
   "FilterTestMessages": true,
   "MoveWSMessages": true,
+  "AutomatedAnalysisConfiguration": {
```
Comment (as2388) [inline configuration/pipeline_config.json]:
```suggestion
  "AutomatedAnalysis": {
```
(It's obvious that this is configuration)

Comment (as2388) [inline configuration/pipeline_config.json]:
```suggestion
  "AutomatedAnalysis": {
```
(It's obvious that this is configuration)

File: src/lib/pipeline_configuration.py [hunk 3]
```diff
@@ -445,3 +451,37 @@ def validate(self):
         validators.validate_string(self.messages_upload_path, "messages_upload_path")
         validators.validate_string(self.individuals_upload_path, "individuals_upload_path")
         validators.validate_string(self.automated_analysis_dir, "automated_analysis_dir")
+
+class AutomatedAnalysisConfiguration(object):
+    def __init__(self, generate_region_theme_distribution_maps, generate_district_theme_distribution_maps,
+                 generate_mogadishu_theme_distribution_maps):
+        """
+        :param generate_region_theme_distribution_maps: whether to generate somali region theme distribution maps.
```
Comment (as2388) [inline src/lib/pipeline_configuration.py]:
```suggestion
        :param generate_region_theme_distribution_maps: Whether to generate somali region theme distribution maps.
```

**Review by as2388 — APPROVED** [review]
approved with a rename of automated_analysis_configuration -> automated_analysis

File: src/lib/pipeline_configuration.py (hunk 3 above)
Comment (IsaackMwenda) [reply src/lib/pipeline_configuration.py]: updated thanks

File: configuration/pipeline_config.json (hunk 2 above)
Comment (IsaackMwenda) [reply configuration/pipeline_config.json]: updated thanks

Comment (IsaackMwenda) [reply configuration/pipeline_config.json]: updated thanks

**Pull request merged**